packages/server-wallet/src/db/migrations/20210126122746_add_objective_waiting_for_column.ts

```diff
@@ -0,0 +1,14 @@
+import * as Knex from 'knex';
+
+const tableName = 'objectives';
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.alterTable(tableName, table => {
+    table.string('waiting_for').notNullable();
```

Comment:
This migration will fail in production. A new column cannot be `notNullable` unless a default value is provided.

Reply:
Good catch. Relevant to https://github.com/statechannels/project-management/issues/55.

Reply:
I made the column nullable in e4cfddd

Reply:
Outside the scope of this PR, but at some point we might want to automate testing of the schema migrations against existing data at some point.

Reply:
Is that consistent with the changeset of this PR, which I think expects `waiting_for` to be `NOT NULL`? (checking for myself in a bit)

Reply:
You're right. The ORM asserts that `waitingFor` is a string, whereas as of that commit it can now be `undefined`. That should be resolved before merging this PR.

Reply:
I think it is best to make `waitingFor` an optional property of a DBObjective. I think that's a better default than `''` or any other string.

Reply:
83bf115

Reply:
Does that mean that in #3301 we'd be returning `undefined` instead of `WaitingFor.nothing`?

Reply:
yes. We would only be doing that when the objective is terminated, though.